Objects/genericaliasobject.c

```diff
@@ -180,14 +180,16 @@ tuple_extend(PyObject **dst, Py_ssize_t dstindex,
 PyObject *
 _Py_make_parameters(PyObject *args)
 {
-    Py_ssize_t nargs = PyTuple_GET_SIZE(args);
+    assert(PyTuple_Check(args) || PyList_Check(args));
+    const bool is_args_tuple = PyTuple_Check(args);
+    Py_ssize_t nargs = is_args_tuple ? PyTuple_GET_SIZE(args) : PyList_GET_SIZE(args);
     Py_ssize_t len = nargs;
     PyObject *parameters = PyTuple_New(len);
     if (parameters == NULL)
         return NULL;
     Py_ssize_t iparam = 0;
     for (Py_ssize_t iarg = 0; iarg < nargs; iarg++) {
-        PyObject *t = PyTuple_GET_ITEM(args, iarg);
+        PyObject *t = is_args_tuple ? PyTuple_GET_ITEM(args, iarg) : PyList_GET_ITEM(args, iarg);
```

Comment:
This is still unsafe for lists, since below you invoke code that might call back into arbitrary Python code that mutates the list. I think for safety, you need to either convert the list into a tuple or use an iteration API.

Reply:
Damn Python getting in the way of my C code :sweat_smile: I'll use the tuple workaround..

Reply:
Ok I think that should be fixed. I now convert the arguments to a tuple before processing them. Btw, do you have any opinion on this? https://github.com/python/cpython/pull/124512#discussion_r1777130931

Reply:
Hi Jelle, just wondering if you've had the time to ponder this issue? What should we do about this PR?

Reply:
Looking at this again I'm not sure what I was thinking of; I think Daraan's post was mostly right.

Reply:
(replying to your [discourse answer](https://discuss.python.org/t/should-list-arguments-of-a-type-statement-that-refer-to-a-paramspec-be-converted-to-a-tuple/66182/2)): I think making generic aliases hashable by converting lists to tuples is a good idea but I'd prefer to implement it in a separate PR (@Daraan already made an issue for it) since it affects both the C (`GenericAlias`) and Python (`_GenericAlias`) implementations.

If that's fine by you, then this PR is ready for a final check :)